auth/scoperesolver.js

```diff
@@ -0,0 +1,338 @@
+var _           = require('lodash');
+var assert      = require('assert');
+var taskcluster = require('taskcluster-client');
+var events      = require('events');
+var base        = require('taskcluster-base');
+var debug       = require('debug')('auth:ScopeResolver');
+
+class ScopeResolver extends events.EventEmitter {
+  /** Create ScopeResolver */
+  constructor() {
+    super();
+
+    // List of client objects on the form:
+    // {clientId, accessToken, expandedScopes: [...], updateLastUsed}
```

Comment:
maybe `updateLastUsedAt` - I assumed this was an async function to update the last-used field.